sdk/keyvault/keyvault-keys/src/cryptography/aesCryptographyProvider.ts

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { OperationOptions } from "@azure/core-http";
+import * as crypto from "crypto";
+import {
+  EncryptOptions,
+  EncryptResult,
+  DecryptOptions,
+  DecryptResult,
+  KeyWrapAlgorithm,
+  WrapKeyOptions,
+  WrapResult,
+  UnwrapKeyOptions,
+  UnwrapResult,
+  SignOptions,
+  SignResult,
+  VerifyOptions,
+  VerifyResult,
+  AesCbcEncryptParameters,
+  JsonWebKey
+} from "..";
+import { AesCbcDecryptParameters } from "../cryptographyClientModels";
+import {
+  CryptographyProvider,
+  CryptographyProviderOperation,
+  LocalCryptographyUnsupportedError
+} from "./models";
+
+/**
+ * An AES cryptography provider supporting AES algorithms.
+ * @internal
+ */
+export class AesCryptographyProvider implements CryptographyProvider {
+  private key: JsonWebKey;
+  constructor(key: JsonWebKey) {
+    this.key = key;
+  }
+  encrypt(
+    encryptParameters: AesCbcEncryptParameters,
+    _options?: EncryptOptions
+  ): Promise<EncryptResult> {
+    const { algorithm, keySizeInBytes } = this.supportedAlgorithms[encryptParameters.algorithm];
+    const iv = encryptParameters.iv || crypto.randomBytes(16);
+
+    this.ensureValid(keySizeInBytes);
+
+    const cipher = crypto.createCipheriv(algorithm, this.key.k!.subarray(0, keySizeInBytes), iv);
```

Comment:
Nit: might separate the params on separate lines like you do below - easier to read and review later.